[WIP] ESM (#2586)
alphagov/govuk-frontend: CHANGELOG.md

## Unreleased

### New features

#### Import GOV.UK Frontend JavaScript as ES modules

> You can now import GOV.UK Frontend's component JavaScript into your service as ES modules, if you're using a bundler.

Reviewer: ```suggestion
You can now import our component JavaScript into your service as ES modules, if you're using a bundler.
```

This allows you to import only the JavaScript you need, and helps reduce duplication of things like polyfills.

> Because we are shipping ES modules in addition to how we currently publish our component JavaScript, this change is

Reviewer: ```suggestion
As we're shipping ES modules in addition to how we currently publish our component JavaScript, this change is
```
P.S. Noob question: what does "in addition to" mean here?

Author: Not a noob question at all. We currently ship our JavaScript in one format (called UMD). This PR introduces a second way of shipping our JavaScript (in a different format, called ESM). What I was trying to say here is that we will continue to ship our JavaScript as UMD. The ESM format is *in addition to* what we're already doing, rather than replacing it.

Reviewer: Ahh I see - ta!

I wonder will we have any users who'll be using our release notes for the first time, and are unaware of how we've done things before.

First go at wording tweak:

> As we're now shipping ES modules in addition to how we already ship JavaScript (using UMD, or Universal Module Definition), this change is backwards compatible.

> backwards compatible - you will not be required to make any changes unless you want to.

Reviewer: ```suggestion
backwards compatible. You will not be required to make any changes unless you want to.
```

> If you want to import using ES modules, we recommend you use `import` to only import the JavaScript for components you're using in your service. For example:

Reviewer: ```suggestion
To import with ES modules, use `import` to only import the JavaScript for components you're using in your service. For example:
```

```javascript
import { SkipLink, Radios } from 'govuk-frontend'

var $skipLink = document.querySelector('[data-module="govuk-skip-link"]')
if ($skipLink) {
  new SkipLink($skipLink).init()
}

var $radios = document.querySelectorAll('[data-module="govuk-radios]')
if ($radios) {
  for (var i = 0; i < $radios.length; i++) {
    new Radios($radios[i]).init()
  }
}
```

If you need to import all of GOV.UK Frontend's components, then run the `initAll` function to initialise them:

```javascript
import { initAll } from 'govuk-frontend'
initAll()
```

> Depending on the bundler you use, you may also need to make changes to your JavaScript bundler configuration file. You can [read more in our installation instructions about importing JavaScript using a bundler](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-javascript-using-a-bundler)

Reviewer: ```suggestion
Depending on the bundler you use, you may also need to make changes to your JavaScript bundler configuration file. [Read more in our installation instructions about importing JavaScript using a bundler](https://frontend.design-system.service.gov.uk/importing-css-assets-and-javascript/#import-javascript-using-a-bundler).
```

> This was added in [pull request #2586: Publish our JavaScript as ESM alongside UMD](https://github.com/alphagov/govuk-frontend/pull/2586).

Reviewer: ```suggestion
This was added in [pull request #2586: Publish our JavaScript as ES Modules alongside Universal Module Definition (UMD)](https://github.com/alphagov/govuk-frontend/pull/2586).
```
